src/backend/distributed/commands/schema.c

```diff
@@ -186,3 +241,41 @@ FilterDistributedSchemas(List *schemas)
 
 	return distributedSchemas;
 }
+
+
+/*
+ * EnsureSequentialModeForSchemaDDL makes sure that the current transaction is already in
+ * sequential mode, or can still safely be put in sequential mode, it errors if that is
+ * not possible. The error contains information for the user to retry the transaction with
+ * sequential mode set from the begining.
+ *
+ * Copy-pasted from type.c
+ */
+static void
+EnsureSequentialModeForSchemaDDL(void)
+{
+	if (!IsTransactionBlock())
+	{
+		/* we do not need to switch to sequential mode if we are not in a transaction */
+		return;
+	}
+
+	if (ParallelQueryExecutedInTransaction())
+	{
+		ereport(ERROR, (errmsg("cannot create or modify schema because there was a "
+							   "parallel operation on a distributed table in the "
+							   "transaction"),
+						errdetail("When creating or altering a schema, Citus needs to "
+								  "perform all operations over a single connection per "
+								  "node to ensure consistency."),
+						errhint("Try re-running the transaction with "
+								"\"SET LOCAL citus.multi_shard_modify_mode TO "
+								"\'sequential\';\"")));
+	}
```

Comment:
lets try and add a test that exhibits this code path.

```sql
BEGIN;
-- A query that will fan out to multiple shards at once, something like a `count(*)` (might need to set a setting to force the count(*) to use maximum parallelism
-- Rename a schema here, this should error
ROLLBACK;
```

Then the same block, however after the begin first run
```sql
SET LOCAL citus.multi_shard_modify_mode TO 'sequential';
```

This makes sure we know that the message works!